metric-example/query-by-example.md

```diff
@@ -0,0 +1,516 @@
+
+For that use case, we want to build a list of averages rather than a list of original samples, and extract the maximum from these averages.
+
+We can take advantage of the fact that `:splitBy` collects pre-aggregated input values (e.g. averages) into a list if the next aggregation after the merge requires it. If we pass all dimensions of the metric to `:splitBy,` no dimensions will be removed and hence no actual merging will happen, instead we have a list with a single average for each time slots. Before the final aggregation, we need to insert `:fold` so that the lists are merged into one over time. From this merged list, we finally extract the maximum with `:max`. Taking all of this together we end up with this selector:
+```
+builtin:host.cpu.usage:avg:splitBy(dt.entity.host):fold:max
+```
+The result is exactly what we wanted: the maximum of average values over time - We can draw the maximum into the chart and see that it will exactly touch the peak of the graph.
+
+If you query this selector together with `builtin:host.cpu.usage:avg` you see that the single value is always exactly equal to the maximum entry of the corresponding series.
+
+# Scenario 15: Filtering for Special Characters with Quoting
+
+>**Task**
+>We have a filter text field, but the resulting metric selector is invalid if we type in special characters.
+
+Some characters have a special meaning in metric selectors: ( ) : , ~ "
+
+Additionally, white space is used to tell where an identifier such as "dt.entity.host" ends. Replacing the periods (full stops) with spaces would result in three identifiers next to each other, rather than a single one that contains white space.
+
+If white space or any of mentioned characters are to be used within a dimension name, in a filter matcher reference, or otherwise in their "literal" sense, rather than as syntax, we need to be careful and escape or quote them. Knowing about proper escaping is especially important when we build our selectors in code that handles user input. In such scenarios, e.g. when a user can provide a filter reference in a text field, we want to prevent them from making the resulting selector syntactically invalid or even worse, allow them to change its meaning and insert a malicious piece of code into our selector.
+
+Say, our user legitimately wants to filter for a host with name `Host 14.A "Alice" ~80° (Europe, North Africa)`. This filter reference can not be inserted into the selector as it is, since it contains a period, double quotes, a tilde, brackets, spaces and a comma. In order not to understand these characters and syntax, we surround the string with double quotes, and escape the contained double quotes with a tilde to indicate that the quoted part is not over yet, but that we have an instance of a literal quote:

```

Comment:
```suggestion
Say, our user legitimately wants to filter for a host with name `Host 14.A "Alice" ~80° (Europe, North Africa)`. This filter reference can not be inserted into the selector as it is, since it contains a period, double quotes, a tilde, brackets, spaces and a comma. In order not to understand these characters as syntax, we surround the string with double quotes and escape the contained double quotes with a tilde to indicate that the quoted part is not over yet, but that we have an instance of a literal quote:

```